### DISCUSSION TITLE
一个小问题：到底有多少软件包只存在于archlinuxcn源里？

### DISCUSSION THREAD

**Discussion by DZ-IO** [archlinuxcn/repo#2770] [Q&A]
官方说明里有这么一条

> 然而并不是所有 [archlinuxcn] 提供的软件包都存在于 AUR 

所以这种只能在archlinuxcn里下到的软件包占了多少？都是啥软件？以及为什么他们无法在aur里下到？

**Comment by lilydjwg** [comment]
这个很难自动化地准确统计，因为有些包可以从 AUR 下 PKGBUILD，然后自己再改一下。准确的办法是比对 PKGBUILD，但是一下子下载一千个 AUR 包的 PKGBUILD 会对 AUR 服务器产生太大压力的吧？谁有兴趣可以写个脚本慢慢跑。

至于为什么不在 AUR 里，我的包的话，一部分是懒得另外维护 AUR 包，比如各种带 -lily 后缀的修改版。另一部分原因是打包方式和 AUR 对应的包并不一致。还有一部分是被 AUR 赶出来了，比如 systemd-rc-local。python-certifi-fakesystem 我甚至都不尝试了。

**Comment by hubutui** [comment]
我曾经添加了大量的 R 包，这些包有的在 AUR 且维护者不是我，有的不在。为了避免麻烦，我就没有往 AUR 推。
但是这部分包目前已经转移到 [BioArchLinux](https://github.com/BioArchLinux/Packages) 那边去了。

我维护的其他包，一般 AUR 上都有对应的，但是 AUR 上的包不一定是我。如果 AUR 上的同名包不是我的话，可能 PKGBUILD 文件也是不完全一致的。

**Reply by Janakrishna394** [reply]
bece7fbe7feb

**Comment by oldherl** [comment]
我的话，有的是因为打包方式依赖于lilac（如qt5-base-24h）无法作为aur使用，有的是自己的修改预计不适用于aur上的大多数用户，还有的是aur上有同名包但我不喜欢它的打包方式而自己修改过了。

**Comment by Y7n05h** [comment]
ArchLinuxCN 和 AUR 的同名包里面虽然有挺多是把 AUR 作为上游去自动构建的，但也还有挺多的包使用了和 AUR 不同的 PKGBUILD。有些可能只是 AUR 的包缺少依赖或者存在一些编译问题，有些问题 ArchLinuxCN 这边的维护者修复了之后可能没在 AUR 反馈（~~我就是在 AUR 反馈的比较少，对同一个包指出好几个问题，好像给他找茬似的，AUR 又不是 Github，开个 PR 多方便~~）。
我还有几个包是 ArchLinuxCN 有而 AUR 上没有的，原因分别有：修改了 AUR 的命名( #2752 )、一组拆分包中的一个在 AUR 上名字被占用（activitywatch）。

**Reply by hubutui** [reply]
哈哈哈，还有人非要在 AUR 上给我各种回复，我没来得及更新，或者更新了各种不满意，非要说我态度不好，一定要求他们在 clean chroot 里构建，或者来这边装二进制包．我都好无语．
